lmdeploy/pytorch/messages.py

```diff
@@ -35,14 +35,18 @@ def __hash__(self):
     def from_gen_config(self, gen_config: EngineGenerationConfig):
         """from gen config."""
 
+        stop_words = gen_config.stop_words or []
+        bad_words = gen_config.bad_words or []
+        if gen_config.ignore_eos:
+            bad_words += stop_words
```

Comment:
No need to clear stop_words? I mean if a word is both a bad_word and a stop_word, what is the expected behavior?

Reply:
https://github.com/InternLM/lmdeploy/blob/aaf3942a5851add8a5bf30b51280ccbb181c3219/lmdeploy/pytorch/engine/logits_process.py#L63-L68

bad words would be unique before sampling.